### PR TITLE
[API Compatibility] Support `optimizer` for `LambdaDecay` and add alias for `DataLoader`

### PR DESCRIPTION
**Pull request by algorithm1832**
### PR Category
User Experience

### PR Types
Improvements

### Description

- Support arg `optimizer` for `LambdaDecay`
- Add alias `paddle.utils.data.DataLoader` for `paddle.io.DataLoader`

**Used AI Studio**

### 是否引起精度变化
否

### CODE REVIEW

File: python/paddle/utils/data/dataloader.py [hunk 1]
```diff
@@ -11,10 +11,78 @@
...
+    from paddle.io.dataloader.dataset import Dataset
+    from paddle.io.reader import _CollateFn
+
+
+class DataLoader(PaddleDataLoader):
+    def __init__(
+        self,
+        dataset: Dataset[Any],
+        batch_size: int | None = 1,
+        shuffle: bool | None = None,
```
Comment (risemeup1111) [inline python/paddle/utils/data/dataloader.py]: P1 **优先级：P1**
`shuffle` 默认成 `None` 后，`paddle.utils.data.DataLoader(dataset)` 会把 `None` 继续传给底层 `paddle.io.DataLoader`。在未提供 `batch_sampler` 的常见路径里，底层会创建 `BatchSampler(..., shuffle=shuffle)`，而 `BatchSampler` 明确断言 `shuffle` 必须是 `bool`，所以默认构造会直接失败。请保持与 `paddle.io.DataLoader`/PyTorch 兼容的默认值 `False`。

```suggestion
        shuffle: bool = False,
```

File: python/paddle/utils/data/dataloader.py [hunk 2]
```diff
@@ -11,10 +11,78 @@
...
+            warnings.warn(
+                "pin_memory, multiprocessing_context, generator, prefetch_factor, pin_memory_device, in_order are currently not supported in DataLoader and will be ignored."
+            )
+
+        if sampler is not None:
+            if batch_sampler is not None:
+                raise ValueError(
+                    "Cannot specify both sampler and batch_sampler"
+                )
+            batch_sampler = sampler
```
Comment (risemeup1111) [inline python/paddle/utils/data/dataloader.py:75]: P1 **优先级：P1**
这里不能把 `sampler` 直接赋给底层 `batch_sampler`。`paddle.io.DataLoader` 的 `batch_sampler` 需要每次 yield 一组样本索引；而 `sampler` 兼容的是 PyTorch 的样本级 sampler，通常每次只 yield 一个 `int`。当前实现会让 fetch 阶段执行 `for idx in batch_indices` 时尝试遍历整数，`DataLoader(dataset, sampler=SequentialSampler(dataset), batch_size=...)` 这类标准用法会失败。请把样本级 sampler 包成 `paddle.io.BatchSampler`，并在使用 `sampler` 时按语义禁止同时设置 `shuffle`。

建议形态：
```python
        if sampler is not None:
            if batch_sampler is not None:
                raise ValueError("Cannot specify both sampler and batch_sampler")
            if shuffle:
                raise ValueError("Cannot specify both sampler and shuffle")
            batch_sampler = paddle.io.BatchSampler(
                sampler=sampler,
                batch_size=batch_size if batch_size is not None else 1,
                drop_last=drop_last,
            )
            batch_size = 1
            drop_last = False
```

**Review by risemeup1111 — CHANGES_REQUESTED** [review]
发现两个需要先修复的兼容性问题，都会影响新增的 `paddle.utils.data.DataLoader` 常见用法。具体建议已放在行级评论里。

CI 目前仍有构建/测试任务在运行；已通过的早期检查未覆盖上述行为路径。

<div align="right">
   <sup>Powered by Nyanpasu with gpt-5.5 xhigh, please check the suggestions carefully.</sup>
</div>

**Review by risemeup1111 — COMMENTED** [review]
已复查最新提交：`shuffle` 默认值问题已经修复。

仍有一个兼容性问题需要处理：`sampler` 仍然直接传给底层 `batch_sampler`，这个问题已在既有行级评论中说明，这里不重复开新线程。CI 目前还有构建/测试任务在运行。

<div align="right">
   <sup>Powered by Nyanpasu with gpt-5.5 xhigh, please check the suggestions carefully.</sup>
</div>

**Review by zhwesky2010 — APPROVED** [review]
